src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductRepository.php

```diff
@@ -258,6 +259,16 @@ public function findByTaxon(TaxonInterface $taxon): array
         ;
     }
 
+    public function findByProductTaxonIds(array $ids): array
+    {
+        return $this->createQueryBuilder('o')
+            ->innerJoin('o.productTaxons', 'productTaxon', Join::WITH, 'productTaxon.id IN (:ids)')
+            ->setParameter('ids', $ids)
```

Comment:
Did you test how it behaves with an empty array?

Reply:
With an empty set it returns nothing, might have been problematic if we put a null into the `:ids` parameter since it gets evaluated differently, but it's strictly typed to an array, so that's not the case.